addon/index.js

```diff
@@ -1,42 +1,49 @@
+/* globals self */
+
 import Ember from 'ember';
 
-let ComparableMoment = Ember.Object.extend(Ember.Comparable, moment.fn, {
+const moment = self.moment;
+
+const ComparableMoment = Ember.Object.extend(Ember.Comparable, moment.fn, {
   compare(a, b) {
-    if (moment.isMoment(a) && moment.isMoment(b) && a.isBefore(b)) {
-      return -1;
-    } else if (moment.isMoment(a) && moment.isMoment(b) && a.isAfter(b)) {
-      return 1;
-    } else if (moment.isMoment(a) && !moment.isMoment(b)) {
-      return 1;
-    } else if (moment.isMoment(b)) {
-      return -1;
+    if (moment.isMoment(a) && moment.isMoment(b)) {
+      if (a.isBefore(b)) {
```

Comment:
The other logic didn't make sense and there were not tests around this so unsure what the original author meant by it.  In any case, implemented logic that made more sense to me and wrote tests.